### PR TITLE
Reorganized the application observability pages

### PR DESCRIPTION
**Pull request by phil-abb**
### Description

Split up the application observability content across the new persona-based sections introduced with the [PR](https://github.com/margo/specification/pull/19) from @ajcraig 

This change only moves content around. There are some textual edits to account for splitting the content up but nothing related to requirements.

The content was split across the new pages indicated below:
image

#### Issues Addressed

N/A

#### Change Type

Please select the relevant options:

- [ ] Fix (change that resolves an issue)
- [ ] New enhancement (change that adds specification content)
- [X] Content edits (change that edits existing content)

#### Checklist

- [X] I have read the [CONTRIBUTING](../CONTRIBUTING.md) document.
- [X] My changes adhere to the established patterns, and best practices.

### CODE REVIEW

**Comment by phil-abb** [comment]
@margo/technical-wg I created a new PR reorganizing the application observability content if anyone wants to review it and provide feedback.

**Review by ajcraig — APPROVED** [review]
Reorg look great, easier to digest by persona.

**Comment by phil-abb** [comment]
@margo/technical-wg This is just a reminder about this PR needing at least one more person to review it. The PR is just moving content around and not changing any of the OTEL requirements.

File: system-design/app-interoperability/publishing-application-observability-data.md
Comment (arne-broering) [inline system-design/app-interoperability/publishing-application-observability-data.md:1]: correct: "is it" to "it is"

File: system-design/device-interoperability/collecting-application-observability-data.md
Comment (arne-broering) [inline system-design/device-interoperability/collecting-application-observability-data.md:1]: "The device owner MUST NOT require the use the sidecar deployment model at this time"
should be:
"The device owner MUST NOT require the use of the sidecar deployment model at this time, "

**Review by arne-broering — APPROVED** [review]

File: system-design/app-interoperability/publishing-application-observability-data.md
Comment (phil-abb) [reply system-design/app-interoperability/publishing-application-observability-data.md:1]: fixed.

File: system-design/device-interoperability/collecting-application-observability-data.md
Comment (phil-abb) [reply system-design/device-interoperability/collecting-application-observability-data.md:1]: fixed.

**Pull request merged**